server/plugin/oAuth.go

```diff
@@ -131,6 +133,50 @@ func (p *Plugin) GenerateOAuthToken(code, state string) error {
 		"redirect_uri":          {fmt.Sprintf("%s%s%s", p.GetSiteURL(), p.GetPluginURLPath(), constants.PathOAuthCallback)},
 	}
 
+	if err := p.GenerateAndStoreOAuthToken(mattermostUserID, oauthTokenFormValues); err != nil {
+		return err
+	}
+
+	if _, err := p.DM(mattermostUserID, fmt.Sprintf("%s\n\n%s", constants.UserConnected, constants.HelpText)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// RefreshOAuthToken refreshes OAuth token
+func (p *Plugin) RefreshOAuthToken(mattermostUserID string) error {
+	user, err := p.Store.LoadUser(mattermostUserID)
+	if err != nil {
+		p.DM(mattermostUserID, constants.GenericErrorMessage)
```

Comment:
You should check error returned from this DM function. If you are logging the error inside the DM function itself, then its return type should not contain error.